### PR TITLE
AWS deployment; Travis Continuous Integration

### PR DESCRIPTION
**Pull request by pierrealixt**
Continous Integration strategy:
- a Pull Request on the branch develop will run the tests.
- a Push on the branch develop will run the tests and deploy to staging. (can be deactivated later)
- a Pull Request on the branch master will run the tests.
- a Push on the branch master will run the tests and deploy to production.

I created one application and two environments on AWS Beanstalk (HOTOSM account). They are configured yet but as soon as we push to develop, they will be automatically configured (wsgi path, env variables).
There are 9 environment variables used by the application:
- APP_SETTINGS: app_config.AWSStagingConfig || app_config.AWSProductionConfig
- OAUTH_CONSUMER_KEY
- OAUTH_SECRET
- ATTIC_DATA_SERVER_URL
- DEFAULT_OVERPASS_URL
- SECRET_KEY
- SENTRY_DSN
- AWS_ACCESS_KEY_ID
- AWS_SECRET_ACCESS_KEY

I created 5 S3 buckets on HOTOSM account:
- hotosm-fieldcampaigner-data-test: used by Travis to run the tests.
- hotosm-fieldcampaigner-data-staging: used by the staging beanstalk environment.
- hotosm-fieldcampaigner-data-production: used by the production beanstalk environment.
- hotosm-fieldcampaigner-develop-deployments: used by Travis to store zip deployment files for the branch develop.
- hotosm-fieldcampaigner-master-deployments: used by Travis to store zip deployment files for the branch master.

In Travis, on the repo hotosm/field-campaigner, I added 4 environment variables:
- AWS_ACCESS_KEY_ID: mapcampaigner IAM user access_key
- AWS_SECRET_ACCESS_KEY: mapcampaigner IAM user secret_access_key
- OAUTH_CONSUMER_KEY: tests fail without it.
- OAUTH_SECRET: tests fail without it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -53,6 +47,31 @@ before_script:
...
+    access_key_id: $AWS_ACCESS_KEY_ID
+    secret_access_key: $AWS_SECRET_ACCESS_KEY
+    region: "us-west-2"
+    app: "field-campaigner"
+    env: "staging"
+    bucket_name: hotosm-fieldcampaigner-develop-deployments
+    on:
+      branch: develop
+
+deploy:
```
Comment (cchristelis) [inline .travis.yml]: duplicate?

File: .travis.yml [hunk 2]
```diff
@@ -13,30 +13,24 @@ compiler: clang
...
-  - '3.5'
-  - '3.4'
 
 env:
   global:
     - ON_TRAVIS=true
     - SECRET_KEY='tT\xd7\xb06\xf7\x9b\xff\x0fZL\xca\xca\x11\xefM\xacr\xfb\xdf\xca\x9b'
-    - APP_SETTINGS=app_config.DevelopmentConfig
-    - DATABASE_URL=localhost
+    # - APP_SETTINGS=app_config.TestingConfig
```
Comment (cchristelis) [inline .travis.yml]: remove commented out code.

File: flask_project/campaign_manager/views.py [hunk 3]
```diff
@@ -692,14 +697,16 @@ def find_attribution(map_url):
 
 @campaign_manager.route('/create', methods=['GET', 'POST'])
 def create_campaign():
+    from datetime import datetime
+    print('create_campaign')
+    print(datetime.now().time())
```
Comment (cchristelis) [inline flask_project/campaign_manager/views.py]: do we need this print outside of a testing environment?

Comment (pierrealixt) [reply flask_project/campaign_manager/views.py]: we don't.

**Pull request merged**